torch/serialization.py

```diff
@@ -604,17 +648,52 @@ def persistent_load(saved_id):
         raise RuntimeError("Invalid protocol version: %s" % protocol_version)
 
     _sys_info = pickle_module.load(f, **pickle_load_args)
-    unpickler = pickle_module.Unpickler(f, **pickle_load_args)
-    unpickler.persistent_load = persistent_load
-    result = unpickler.load()
+
+    # Read through the data pickle without actually loading anything, just
+    # continue through it until a STOP opcode
+    data_start = f.tell()
+
+    # Skip until the next op with `expected_name` is seen
+    def get_next(expected_name, f):
+        op, value, _ = next(pickletools.genops(f))
+        while op.name != expected_name:
+            op, value, _ = next(pickletools.genops(f))
+        return value
+
+    storage_sizes = {}
+
+    # This small pickle implementation searches through the file for tensors
+    # to find the size of each tensor so they can be eagerly loaded when the
+    # real pickle load is called. Using the storage type and numel, each tensor's
+    # size can be found.
+    for op, value, _ in pickletools.genops(f):
+        if op.name == 'GLOBAL':
+            if 'torch._utils _rebuild_tensor_v2' in value:
```

Comment:
you'd also need to support qtensor (can it be abstracted into a function with corresponding comments?)

also - is this space intended?